### PR TITLE
`sage --package metrics`: New tool to assist discussions of the Sage distribution

### PR DESCRIPTION
**Pull request by mkoeppe**
Discussions of the complexity of the Sage distribution pop up unexpectedly, as seen in

https://github.com/sagemath/sage/pull/36982#issuecomment-1873202140, https://github.com/sagemath/sage/pull/36982#issuecomment-1874285785, https://github.com/sagemath/sage/pull/36982#issuecomment-1874328804, https://github.com/sagemath/sage/pull/36982#issuecomment-1874446757, https://github.com/sagemath/sage/pull/36982#issuecomment-1874452301, https://github.com/sagemath/sage/pull/36982#issuecomment-1874454350, https://github.com/sagemath/sage/pull/36982#issuecomment-1874557090

https://github.com/sagemath/sage/pull/36726#issuecomment-1821618629, https://github.com/sagemath/sage/pull/36726#issuecomment-1821795302, https://github.com/sagemath/sage/pull/36726#issuecomment-1836736491, https://github.com/sagemath/sage/pull/36726#issuecomment-1837111688, https://github.com/sagemath/sage/pull/36726#issuecomment-1838189407, https://github.com/sagemath/sage/pull/36726#issuecomment-1838759510, https://github.com/sagemath/sage/pull/36726#issuecomment-1839341903, https://github.com/sagemath/sage/pull/36726#issuecomment-1839596640, https://github.com/sagemath/sage/pull/36726#issuecomment-1840193953, https://github.com/sagemath/sage/pull/36726#issuecomment-1840362382, https://github.com/sagemath/sage/pull/36726#issuecomment-1843422890, https://github.com/sagemath/sage/pull/36726#issuecomment-1843584828, https://github.com/sagemath/sage/pull/36726#issuecomment-1843760600, https://github.com/sagemath/sage/pull/36726#issuecomment-1846117906, https://github.com/sagemath/sage/pull/36726#issuecomment-1848641519, https://github.com/sagemath/sage/pull/36726#issuecomment-1871890028, https://github.com/sagemath/sage/pull/36726#issuecomment-1872063393

https://github.com/sagemath/sage/pull/36777#issuecomment-1832416522, https://github.com/sagemath/sage/pull/36777#issuecomment-1832818351, https://github.com/sagemath/sage/pull/36777#issuecomment-1833364886, https://github.com/sagemath/sage/pull/36777#issuecomment-1836881844, https://github.com/sagemath/sage/pull/36777#issuecomment-1837100711, https://github.com/sagemath/sage/pull/36777#issuecomment-1837426485

To help put such discussions on a solid factual basis, we introduce the command `sage --package metrics`.

```
$ ./sage -package metrics :standard:
has_file_distros_arch_txt=131
has_file_distros_conda_txt=216
has_file_distros_debian_txt=125
has_file_distros_fedora_txt=138
has_file_distros_gentoo_txt=181
has_file_distros_homebrew_txt=61
has_file_distros_macports_txt=129
has_file_distros_nix_txt=51
has_file_distros_opensuse_txt=146
has_file_distros_slackware_txt=25
has_file_distros_void_txt=184
has_file_patches=35
has_file_spkg_check=59
has_file_spkg_configure_m4=222
has_file_spkg_install=198
has_tarball_upstream_url=231
line_count_file_patches=22561
line_count_file_spkg_check=402
line_count_file_spkg_configure_m4=2792
line_count_file_spkg_install=2960
packages=272
type_standard=272
```

Use `PATH=build/bin:$PATH SAGE_ROOT=some-other-worktree build/bin/sage-package metrics :standard:` to obtain the metrics of another version of Sage in some other worktree.

We add computation and before/after comparison of the metrics to the CI Linux Incremental workflow.
As an illustration, we change one Python package from "normal" to "wheel", removing an `spkg-install.in` file in the process. See https://github.com/sagemath/sage/actions/runs/7342841283/job/19992606617?pr=36977#step:6:12

More metrics can be added after
- #36740

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by kwankyu** [comment]
You may want to document `--package metrics` in `https://doc.sagemath.org/html/en/developer/packaging.html`. In particular, please explain the meanings of exemplary output lines. Some of them are not obvious unless you read the code.

**Comment by mkoeppe** [comment]
Thanks for the suggestion. Done in 6e0d890

File: src/doc/en/developer/packaging.rst [hunk 1]
```diff
@@ -412,19 +412,19 @@ begin with ``sdh_``, which stands for "Sage-distribution helper".
...
 
 - ``sdh_setup_bdist_wheel [...]``: Runs ``setup.py bdist_wheel`` with
-   the given arguments, as well as additional default arguments used for
-   installing packages into Sage.
+  the given arguments, as well as additional default arguments used for
+  installing packages into Sage.
 
-- ``sdh_pip_install [...]``: The equivalent of running ``pip install``
+-  ``sdh_pip_install [...]``: The equivalent of running ``pip install``
    with the given arguments, as well as additional default arguments used for
```
Comment (kwankyu) [inline src/doc/en/developer/packaging.rst]: Two spaces, unlike the previous paragraph?

**Comment by kwankyu** [comment]
Otherwise, lgtm.

**Comment by kwankyu** [comment]
The PR description seems screwed up.

File: src/doc/en/developer/packaging.rst (hunk 1 above)
Comment (mkoeppe) [reply src/doc/en/developer/packaging.rst]: Fixed in 6731a1c

**Comment by mkoeppe** [comment]
Thanks!

**Pull request merged**